server/services/allocations.ts

```diff
@@ -57,3 +57,25 @@ export const updateAllocation = async (allocationId: number, allocation, user: H
   const res = await dbClient.db[collections.SITE_PHASE_ALLOCATION].update(allocationId, data);
   return res;
 };
+
+export const createBulkAllocation = async (payload: any, user: HcapUserInfo) => {
+  const updateResults = [];
+  await Promise.all(
+    payload.siteIds.map(async (id) => {
+      const allocation = await getAllocation(id, payload.phase_id);
```

Comment:
This feels like a lot of requests as well. Can we write up a service for getting allocation for multiple sites in a phase? Or maybe grabbing all and filtering based on siteId?

Reply:
good call - I have updated it to make 1 query to `getAllocationsForSites`, I then check to ensure the site being referenced exists in the array or existing allocations `allocationFound`, if it is, I update allocation, if not I create allocation. Let me know if this solution is more aligned with what you were thinking.